iidm/iidm-tck/src/test/java/com/powsybl/iidm/network/tck/AbstractAreaTest.java

```diff
@@ -0,0 +1,489 @@
+    }
+
+    @Test
+    public void testSetterGetterInMultiVariants() {
+        // Changes listener
+        NetworkListener mockedListener = Mockito.mock(DefaultNetworkListener.class);
+        // Set observer changes
+        network.addListener(mockedListener);
+
+        // Init variant manager
+        VariantManager variantManager = network.getVariantManager();
+        List<String> variantsToAdd = Arrays.asList("s1", "s2", "s3", "s4");
+        variantManager.cloneVariant(VariantManagerConstants.INITIAL_VARIANT_ID, variantsToAdd);
+
+        variantManager.setWorkingVariant("s4");
+        // check values cloned by extend
+        assertEquals(-602.6, controlAreaA.getAcInterchangeTarget().orElseThrow(), 0.0);
+        // change values in s4
+        controlAreaA.setAcInterchangeTarget(123.0);
+        // Check P0 & Q0 update notification
+        Mockito.verify(mockedListener, Mockito.times(1)).onUpdate(controlAreaA, "acInterchangeTarget", "s4", -602.6, 123.0);
+
+        // remove s2
+        variantManager.removeVariant("s2");
+
+        variantManager.cloneVariant("s4", "s2b");
+        variantManager.setWorkingVariant("s2b");
+        // check values cloned by allocate
+        assertEquals(123.0, controlAreaA.getAcInterchangeTarget().orElseThrow(), 0.0);
+        // recheck initial variant value
+        variantManager.setWorkingVariant(VariantManagerConstants.INITIAL_VARIANT_ID);
+        assertEquals(-602.6, controlAreaA.getAcInterchangeTarget().orElseThrow(), 0.0);
+
+        // remove working variant s4
+        variantManager.setWorkingVariant("s4");
+        variantManager.removeVariant("s4");
+        try {
+            controlAreaA.getAcInterchangeTarget();
+            fail();
+        } catch (Exception ignored) {
+            // ignore
+        }
+
+        // Remove observer changes
+        network.removeListener(mockedListener);
+    }
+
+    @Test
+    public void testGetAreaBoundary() {
+        AreaBoundary areaBoundary = controlAreaA.getAreaBoundary(dlXnode1A.getBoundary());
+        assertNotNull(areaBoundary);
+        assertTrue(areaBoundary.isAc());
+        assertEquals(controlAreaA.getId(), areaBoundary.getArea().getId());
+        assertEquals(-301.315, areaBoundary.getP(), DELTA);
+        assertEquals(-116.524, areaBoundary.getQ(), DELTA);
+
+        controlAreaA.removeAreaBoundary(dlXnode1A.getBoundary());
+        assertNull(controlAreaA.getAreaBoundary(dlXnode1A.getBoundary()));
+    }
+
+    @Test
+    public void areaInterchangeComputation() {
+        assertEquals(-602.631, controlAreaA.getAcInterchange(), DELTA);
+        assertEquals(0.0, controlAreaA.getDcInterchange());
+        assertEquals(-602.631, controlAreaA.getTotalInterchange(), DELTA);
+
+        assertEquals(+603.563, controlAreaB.getAcInterchange(), DELTA);
+        assertEquals(0.0, controlAreaB.getDcInterchange());
+        assertEquals(+603.563, controlAreaB.getTotalInterchange(), DELTA);
```

Comment:
@olperr1 @annetill @geofjamg 

This is not so nice here, because the two values should be equal of opposite sign.
`Control Area A: -602.631 MW`
`Control Area B: +603.563 MW`

The discrepancy is caused by `EurostagTutorialExample1Factory#createWithTieLines` which introduces a non-zero G for the dangling lines, although `EurostagTutorialExample1Factory#create` has zero G-s for the 2 lines, and the power flow results in here seem to be built using zero G-s.

If I modify `createWithTieLines` with zero G-s for the dangling lines, then I find much better values:
`Control Area A: -602.949 MW`
`Control Area B: +602.944 MW`
Which fits with powerflow tolerances...

Was this G introduction on DanglingLines made on purpose ? Do we need to update the powerflow results specifically for the `createWithTieLines` case ?
Or, should we fix to use zero G-s within this MR or another MR ?

Thanks

Reply:
The non-zero G-s come from #2599. They were set to avoid problems when executing the tests in Java 17 and in Java >= 19, where the results are not exactly the same.
As you suggested, setting the G-s to zero in your specific test case may be a solution. But it should be tested with Java 17 and >=19 versions.

Reply:
Thanks, 
Here what I did in 68dbd546b59ef717b340e9635192129d3dd723b9:
- The previously existing `createWithTieLines` remains unchanged (with non-zero DL G's because of #2599)
- The new `createWithTieLinesAndAreas` introduced in this PR has zero DL G's

Because `createWithTieLinesAndAreas` is not used in any SerDe related test, no issue introduced (tested also locally with Java 19 and Java 21).

Reply:
It's OK for me. Thanks.